src/rpc/net.cpp

```diff
@@ -1016,6 +1016,55 @@ static RPCHelpMan sendmsgtopeer()
     };
 }
 
+static RPCHelpMan getaddrmaninfo()
+{
+    return RPCHelpMan{"getaddrmaninfo",
+                      "\nProvides information about the node's address manager by returning the number of "
+                      "addresses in the `new` and `tried` tables and their sum for all networks.\n"
+                      "This RPC is for testing only.\n",
+                      {},
+                      RPCResult{
+                              RPCResult::Type::OBJ_DYN, "", "json object with network type as keys",
+                              {
+                                      {RPCResult::Type::OBJ, "network", "the network (" + Join(GetNetworkNames(), ", ") + ")",
+                                       {
+                                               {RPCResult::Type::NUM, "new", "number of addresses in the new table, which represent potential peers the node has discovered but hasn't yet successfully connected to."},
+                                               {RPCResult::Type::NUM, "tried", "number of addresses in the tried table, which represent peers the node has successfully connected to in the past."},
+                                               {RPCResult::Type::NUM, "total", "total number of addresses in both new/tried tables"},
+                                       }},
+                              }
+                      },
+                      RPCExamples{
+                              HelpExampleCli("getaddrmaninfo", "")
+                              + HelpExampleRpc("getaddrmaninfo", "")
+                      },
+                      [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+                      {
+                          NodeContext& node = EnsureAnyNodeContext(request.context);
+                          if (!node.addrman) {
+                              throw JSONRPCError(RPC_CLIENT_P2P_DISABLED, "Error: Address manager functionality missing or disabled");
+                          }
```

Comment:
nit: to deduplicate code, could introduce a helper function `EnsureAddrman` in rpc/server_util.cpp (similar to the already existing `EnsureConnman` and `EnsurePeerman`) and use that here and in the `addpeeraddress` RPC.

Reply:
done in #28565.